lib/IRGen/IRGenSIL.cpp

```diff
@@ -757,16 +681,19 @@ class IRGenSILFunction :
 
   /// Account for bugs in LLVM.
   ///
+  /// - When a variable is spilled into a stack slot, LiveDebugValues fails to
+  ///   recognize a restore of that slot for a different variable.
+  ///
   /// - The LLVM type legalizer currently doesn't update debug
   ///   intrinsics when a large value is split up into smaller
   ///   pieces. Note that this heuristic as a bit too conservative
   ///   on 32-bit targets as it will also fire for doubles.
   ///
   /// - CodeGen Prepare may drop dbg.values pointing to PHI instruction.
   bool needsShadowCopy(llvm::Value *Storage) {
-    return (IGM.DataLayout.getTypeSizeInBits(Storage->getType()) >
-            IGM.getClangASTContext().getTargetInfo().getRegisterWidth()) ||
-           isa<llvm::PHINode>(Storage);
+    return !isa<llvm::Constant>(Storage) ||
+           (IGM.DataLayout.getTypeSizeInBits(Storage->getType()) >
+            IGM.getClangASTContext().getTargetInfo().getRegisterWidth());
```

Comment:
but this condition, I'm not sure of

Reply:
I would expect all non-constants to require a shadow copy?

Reply:
All non-constants do get a shadow copy, I left the register-width condition in place to handle constants that don't fit in a register. However it looks like swift IRgen doesn't produce this kind of constant, so I'll just delete the condition.

Reply:
Sorry I misread that. We cannot express constants > 64bits in LLVM IR. A 80-bit float or a 128-bit vector would violate that requirement.

Reply:
For swift, I think it's safe to remove the register-width check, based on a quick skim of GenConstant/IRGenSIL.

In the past though, clang has emitted e.g. i65 integers to implement overflow checks. In some cases it might still use i129 etc.